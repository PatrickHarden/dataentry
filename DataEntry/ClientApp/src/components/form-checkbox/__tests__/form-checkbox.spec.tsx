import React from 'react';
import { mount, shallow } from 'enzyme';

import FormCheckbox from '../form-checkbox';

describe('components', () => {

  describe('form-checkbox', () => {

    let props:any;
    let wrapper:any;

    beforeEach(() => {
        props = {
            name: "testme",
            label: "I am a checkbox",
            defaultValue: true,
            changeHandler: jest.fn()
        }
    });

    // it is a checkbox
    it("it should be a checkbox (type = checkbox)", () => {
        wrapper = mount(<FormCheckbox {...props}/>);
        expect(wrapper.find('input').props().type).toEqual("checkbox")
    });

    // it sets the checkbox as selected by default (field.value = true)
    it("should be checked (with value of true)", () => {
        wrapper = mount(<FormCheckbox {...props}/>);
        expect(wrapper.find('input').props().checked).toEqual(true)
    });

    // it has a label of "I am a checkbox"
    it("renders the expected label", () => {
        wrapper = mount(<FormCheckbox {...props}/>);
        expect(wrapper.find('h5').text()).toEqual("I am a checkbox");
    });

    // snapshot render
    it('renders according to snapshot', () => {
        expect(shallow(<FormCheckbox {...props} />)).toMatchSnapshot();
    });
  });
});