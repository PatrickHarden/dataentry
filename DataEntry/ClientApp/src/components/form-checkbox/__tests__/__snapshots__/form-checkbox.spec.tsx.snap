// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components form-checkbox renders according to snapshot 1`] = `
<styled.div
  changeHandler={[MockFunction]}
  defaultValue={true}
  label="I am a checkbox"
  name="testme"
>
  <styled.input
    checked={true}
    data-testid="form-checkbox"
    id="testme"
    name="testme"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    tabIndex={0}
    type="checkbox"
  />
  <styled.span
    className="checkmark"
    onKeyDown={[Function]}
    outline="0"
  />
  <FormFieldLabel
    label="I am a checkbox"
    styles={Object {}}
  />
</styled.div>
`;
